File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, nums: List[int]) -> int:
         i=0
         n=len(nums)
         j=n-1
         ans=0
         while(i<=j):
            ans=max(ans,min(nums[i],nums[j])*(j-i))
            if(nums[i]<nums[j]):
                i+=1
            else: 
                j-=1
         return ans

File: 11-container-with-most-water/container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
     int ans=0;int j=height.size()-1;
     int i=0;
     while(i<=j){
        ans=max(ans,min(height[j],height[i])*(j-i));
        if(height[i]>height[j]){
            j--;
        }
        else {
            i++;
        }
     }
     return ans;
    }
};